core/src/main/java/com/whylogs/core/ColumnProfile.java

```diff
@@ -31,28 +34,44 @@
 public class ColumnProfile {
   public static final int FREQUENT_MAX_LG_K = 7;
   private static final int CARDINALITY_LG_K = 12;
+  private static volatile Set<String> NULL_STR_ENVS;
 
   @NonNull private final String columnName;
   @NonNull private final CountersTracker counters;
   @NonNull private final SchemaTracker schemaTracker;
   @NonNull private final NumberTracker numberTracker;
   @NonNull private final ItemsSketch<String> frequentItems;
   @NonNull private final HllSketch cardinalityTracker;
+  @NonNull private final Set<String> nullStrs;
+
+  static Set<String> nullStrsFromEnv() {
+    if (ColumnProfile.NULL_STR_ENVS == null) {
+      val nullSpec = System.getenv("NULL_STRINGS");
+      ColumnProfile.NULL_STR_ENVS =
+          nullSpec == null ? Collections.emptySet() : Sets.newHashSet(nullSpec.split(","));
+    }
+    return ColumnProfile.NULL_STR_ENVS;
+  }
 
   public ColumnProfile(String columnName) {
+    this(columnName, nullStrsFromEnv());
+  }
+
+  public ColumnProfile(String columnName, Set<String> nullStrs) {
     this.columnName = columnName;
     this.counters = new CountersTracker();
     this.schemaTracker = new SchemaTracker();
     this.numberTracker = new NumberTracker();
     this.frequentItems = FrequentStringsSketch.create();
     this.cardinalityTracker = new HllSketch(CARDINALITY_LG_K);
+    this.nullStrs = nullStrs;
   }
 
   public void track(Object value) {
     synchronized (this) {
       counters.incrementCount();
 
-      if (value == null) {
+      if (value == null || this.nullStrs.contains(value.toString())) {
```

Comment:
```suggestion
      if (value == null || (!this.nullStrs.isEmpty() && this.nullStrs.contains(value.toString()))) {
```

Not sure if you saw my previous comment but I agree that we shouldn't do `.toString` everywhere

Reply:
right, just saw that, fixing now